packages/devtools_app/lib/src/screens/inspector/diagnostics.dart

```diff
@@ -58,10 +59,56 @@ class DiagnosticsNodeDescription extends StatelessWidget {
     );
   }
 
-  Iterable<TextSpan> _buildDescriptionTextSpans(
+  /// Approximates the width of the elements inside a [RemoteDiagnosticsNode]
+  /// widget.
+  static double approximateNodeWidth(
+    RemoteDiagnosticsNode? diagnostic,
+  ) {
+    final spans = DiagnosticsNodeDescription.buildDescriptionTextSpans(
+      diagnostic?.description ?? '',
+      const TextStyle(),
+      const ColorScheme.dark(),
+      diagnostic,
+      null,
+      null,
+    );
+
+    var spanWidth = spans.fold<double>(
+      0,
+      (sum, span) => sum + calculateTextSpanWidth(span),
+    );
+    String? name = diagnostic?.name;
+
+    // An Icon is approximately the width of 1 character
+    final approximateIconWidth = calculateTextSpanWidth(
+      const TextSpan(
+        text: 'A',
+      ),
+    );
+
+    if (diagnostic?.showName == true && name != null) {
+      // The diagnostic will show it's name instead of an icon so add an
+      // approximate name width.
+
+      if (diagnostic?.description != null) {
+        // If there is a description then a separator will show with the name.
+        name += ': ';
+      }
+      spanWidth += calculateTextSpanWidth(TextSpan(text: name));
+    } else {
+      // When there is no name, an icon will be shown with the text spans.
+      spanWidth += approximateIconWidth;
```

Comment:
if we only need `approximateIconWidth` in this else statement, initialize it here so we save a calculation in the if statement case